docs/workspaces.md

```diff
@@ -60,6 +60,20 @@ they have independent behaviour and no cluster workspace sees objects from other
 cluster workspaces. In contrast to namespace in Kubernetes, this includes non-namespaced
 objects, e.g. like CRDs where each workspace can have its own set of CRDs installed.
 
+## User Home Workspaces
+
+User home workspaces are an optional feature of kcp. If enabled (through `--enable-home-workspaces`), there is a special virtual `Workspace` called `~` in the root workspace. It is used by `kubectl ws` (which similarly to Unix `cd` move the users to their home) to derive the full path to the user home workspace. This full path depends on the user name, the configured number (default 2), the length (default 2) of bucket workspaces and the root prefix (default `root:users`). With that a user `adam` is mapped to e.g. `root:users:a8:f1:adam`. The bucket names are deterministically derived from the user name (via some hash).
```

Comment:
```suggestion
User home workspaces are an optional feature of kcp. If enabled (through `--enable-home-workspaces`), there is a special virtual `Workspace` called `~` in the root workspace. It is used by `kubectl ws` to derive the full path to the user home workspace, similar to how Unix `cd ~` moves a user to their home directory. This full path depends on the user name, the configured number (default 2), the length (default 2) of bucket workspaces and the root prefix (default `root:users`). With that a user `adam` is mapped to e.g. `root:users:a8:f1:adam`. The bucket names are deterministically derived from the user name (via some hash).
```

Reply:
```
The full path for a user's home workspace has a number of parts: `<prefix>:(<bucket>:)+<user-name>`. Buckets are used to ensure that ~1000 sub-buckets or users exist in any bucket, for scaling reasons. The `kcp` administrator can configure:

- the `<prefix>`, which defaults to `root:users`
- the bucket depth, which defaults to 2
- the bucket length, which defaults to 2

Note: with the current setup, ~60 users or ~2000 sub-buckets will be in any bucket. DO NOT set the bucket size to be longer than 2, as this will adversely impact performance.
```

Reply:
cc @sttts @davidfestal 

IMHO - these options are pretty esoteric and an enormous foot-gun. No sane admin would ever choose something other than 2,2 - let's remove the options. This is silly.

Reply:
e.g. choosing bucket length > 2 leads to a performance issue. So, we have the following options:

User-names have (26 * [(26 + 10 + 2) * 61] * 36 = 2169648) permutations, and buckets are made up of lowercase-alpha chars, so:

- length = 1, depth = 2 -> 26 sub-buckets per bucket; 2169648 / (26) ^2 = 3200 users per bucket
- length = 1, depth = 3 -> 26 * 38 = 988 sub-buckets per bucket; 2169648 / (26 * 38)^3 = 124 users per bucket
- length = 2, depth = 1 -> 26 * 38 = 988 sub-buckets per bucket; 2169648 / (26 * 38)^2 = 2196 users per bucket
- length = 2, depth = 2 -> 26 * 38 = 988 sub-buckets per bucket; 2169648 / (26 * 38)^2 = 3 users per bucket

Every single other option is nonsense. We should not have more than ~1000 sub-buckets per bucket, and we don't want fewer than 3 users per bucket .... 

IMHO we hard-code length = 1, depth = 3, and just go with it. Exposing these options is an illusion of choice.

Reply:
If this doc accurately depicts the current situation and if we agree that removing the option is probably the right thing (or at least worth discussing more) should we call this good and create an issue for proposing the default and removal of config option and include this breakdown?

Reply:
(to be clear: I'm happy to create that issue, just give me the 👍 )

Reply:
@pweil- I think we should amend the comments in the doc to something like [this](https://github.com/kcp-dev/kcp/pull/1561#discussion_r926886232) to be more clear

Reply:
For the follow-up: I see value in the flags, but would clearly document the sensible cases in the flag docs, and add the math here (follow-up).

Reply:
follow up: https://github.com/kcp-dev/kcp/issues/1599

I've revamped the whole thing to include better organization, config info, and the example tables.  PTAL 👍